.NET/Samples/BotBuilder_v4/BotBuilderRecognizerBotv4Bot.cs

```diff
@@ -0,0 +1,275 @@
+                        // And finish by sending a message to the user. Next time ContinueAsync is called it will return DialogTurnStatus.Empty.
+                        await turnContext.SendActivityAsync(MessageFactory.Text(results.Result.ToString()),cancellationToken);
+
+                        if (!isDeliverySet)
+                        {
+                            await dialogContext.PromptAsync(
+                            "delivery",
+                            new PromptOptions
+                            {
+                                Prompt = MessageFactory.Text("When do you want to receive the delivery?\n\n" +
+                                "Some valid options are:\n\n" +
+                                " - Tomorrow morning\n\n" +
+                                " - 12/30/2017\n\n" +
+                                " - 9PM Tomorrow\n\n" +
+                                " - Five hours from now"),
+                                RetryPrompt = MessageFactory.Text("I'm sorry, that doesn't seem to be a valid delivery date and time"),
+                            },
+                            cancellationToken);
+                        }
+                    }
+                }
+            }
+            else if (turnContext.Activity.Type == ActivityTypes.ConversationUpdate)
+            {
+                if (turnContext.Activity.MembersAdded != null)
+                {
+                    // Send a welcome message to the user and tell them what actions they may perform to use this bot
+                    await SendWelcomeMessageAsync(turnContext, cancellationToken);
+                }
+            }
+            else
+            {
+                await turnContext.SendActivityAsync($"{turnContext.Activity.Type} event detected", cancellationToken: cancellationToken);
+            }
+
+            // Save the new turn count into the conversation state.
+            await _accessors.ConversationState.SaveChangesAsync(turnContext, false, cancellationToken);
+        }
+
+        /// <summary>
+        /// This is an example of a custom validator. This example can be directly used on a float NumberPrompt.
+        /// Returning true indicates the recognized value is acceptable. Returning false will trigger re-prompt behavior.
+        /// </summary>
+        /// <param name="promptContext">The <see cref="PromptValidatorContext"/> gives the validator code access to the runtime, including the recognized value and the turn context.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>A <see cref="Task"/> representing the operation result of the Turn operation.</returns>
+        public Task<bool> QuantityValidatorAsync(PromptValidatorContext<string> promptContext, CancellationToken cancellationToken)
+        {
+            var result = promptContext.Recognized.Value;
+            promptContext.Recognized.Succeeded = false;
+
+            if (result != null)

```

Comment:
I suggest using a different approach here to handle these conditions. Instead of using:

```
if (conditian_A)
{
     if(condition_B)
     { ... }
     else { return Task.FromResult(false); }
}
else { return Task.FromResult(false); }
```

We can use for improving readability:
```
if (!condition_A)
{
   return Task.FromResult(false);
}

if (!condition_B)
{
    return Task.FromResult(false):
}

// Here goes the rest of the code 
```

Reply:
Feedback applied. Thanks!